db/sql/0_msar.sql

```diff
@@ -130,16 +130,48 @@ $$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
 
 CREATE OR REPLACE FUNCTION
 msar.get_column_name(table_id oid, col_attnum integer) RETURNS text AS $$/*
-Return the name for a given table, qualified or quoted as appropriate.
+Return the name for a given column in a given table.
+
+Args:
+  table_id:  The OID of the table.
+  column_attnum:   The attnum of the column in the table.
+*/
+BEGIN
+  RETURN quote_ident(attname::text) FROM pg_attribute WHERE attrelid=table_id AND attnum=col_attnum;
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
 
-In cases where the table is already included in the search path, the returned name will not be
-fully-qualified.
+
+CREATE OR REPLACE FUNCTION
+msar.is_pkey_col(table_id oid, col_attnum integer) RETURNS boolean AS $$/*
+Return whether the given column is in the primary key of the given table.
+*/
+BEGIN
+  RETURN ARRAY[col_attnum::smallint] <@ conkey FROM pg_constraint WHERE conrelid=table_id;
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
+
+
+CREATE OR REPLACE FUNCTION
+msar.get_cast_function_name(target_type regtype) RETURNS text AS $$/*
+Return a string giving the appropriate name of the casting function for the target_type.
 
 Args:
-  table_id:   The OID of the table.
+  target_type: This should be a type that exists.
 */
+DECLARE target_type_prepped text;
 BEGIN
-  RETURN attname::text FROM pg_attribute WHERE attrelid=table_id AND attnum=col_attnum;
+  WITH unqualifier AS (
```

Comment:
What's the rationale for unqualifying a type? Does this not introduce the possibility for a collission?

Reply:
If this does introduce the possibility for a collission, but you're making the compromise so as to not have to refactor our casting functions, I think this warrants a TODO code comment.